src/EditorFeatures/Core/EditorConfigSettings/Data/CodeStyleSetting.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Updater;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data
+{
+    internal abstract class CodeStyleSetting : Setting
+    {
+        private static readonly bool[] s_boolValues = new[] { true, false };
```

Comment:
consider an IMmutableArray.  Would be bad if anyone mutated these :)

Reply:
Mutable arrays are currently used across EditorConfigSettings, so i'll leave it for another time.